ResearchKit/Common/ORKTextFieldView.m

```diff
@@ -64,6 +67,84 @@ - (CGRect)caretRectForPosition:(UITextPosition *)position {
 @end
 
 
+@implementation ORKPasscodeTextField {
+    NSInteger _numberOfDigits;
+}
+
+- (instancetype)initWithNumberOfDigits:(NSInteger)numberOfDigits {
+    self = [super init];
+    if (self) {
+        _numberOfDigits = numberOfDigits;
+        self.font = [UIFont fontWithName:@"Courier" size:35.0];
+        self.textAlignment = NSTextAlignmentCenter;
+        
+        [self updateTextWithNumberOfFilledBullets:0];
+    }
+    return self;
+}
+
+- (NSInteger)numberOfDigits {
+    return _numberOfDigits;
+}
+
+- (UIKeyboardType)keyboardType {
+    return UIKeyboardTypeNumberPad;
+}
+
+- (BOOL)allowsSelection {
+    return NO;
+}
+
+- (void)updateTextWithNumberOfFilledBullets:(NSInteger)filledBullets {
+    
+    // Error checking.
+    if (filledBullets > _numberOfDigits) {
+        @throw [NSException exceptionWithName:NSInvalidArgumentException
+                                       reason:@"The number of filled bullets cannot exceed the number of pin digits."
+                                     userInfo:nil];
+    }
+    
+    // Append the filled bullets.
+    NSMutableString *text = [NSMutableString new];
+    for (NSInteger i = 0; i < filledBullets; i++) {
```

Comment:
How about use this?
https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSString_Class/#//apple_ref/occ/instm/NSString/stringByPaddingToLength:withString:startingAtIndex: